gtests_dbms build failed. Because StorageDeltaMerge constructor not match.

## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
```
make gtests_dbms -j10
```
Then got
```
In file xxxxxx/dbms/src/Flash/Management/tests/gtest_manual_compact.cpp:20:
In file xxxxxx/dbms/src/Storages/DeltaMerge/tests/MultiSegmentTestUtil.h:25:
In file xxxxxx/dbms/src/Storages/StorageDeltaMerge.h:28:
xxxx/libs/libcommon/include/ext/shared_ptr_helper.h:46:19: error: no matching constructor for initialization of 'DB::StorageDeltaMerge'
                : T(std::forward<TArgs>(args)...){};
                  ^ ~~~~~~~~~~~~~~~~~~~~~~~~~
xxxx/libs/libcommon/include/ext/shared_ptr_helper.h:43:16: note: in instantiation of member function 'ext::SharedPtrHelper<DB::StorageDeltaMerge>::create(char const (&)[8], char const (&)[8], char const (&)[11], std::__1::reference_wrapper<TiDB::TableInfo> &&, DB::ColumnsDescription &&, std::__1::shared_ptr<DB::IAST> &, int &&, DB::Context &)::Local::Local' requested here
        struct Local : T
               ^
xxxx/dbms/src/Flash/Management/tests/gtest_manual_compact.cpp:83:38: note: in instantiation of function template specialization 'ext::SharedPtrHelper<DB::StorageDeltaMerge>::create<char const (&)[8], char const (&)[8], char const (&)[11], std::__1::reference_wrapper<TiDB::TableInfo>, DB::ColumnsDescription, std::__1::shared_ptr<DB::IAST> &, int, DB::Context &>' requested here
        storage = StorageDeltaMerge::create("TiFlash",
                                     ^
xxxx/dbms/src/Storages/StorageDeltaMerge.h:169:5: note: candidate constructor not viable: no known conversion from 'std::__1::reference_wrapper<TiDB::TableInfo>' to 'const DM::OptionTableInfoConstRef' (aka 'const optional<std::reference_wrapper<const TiDB::TableInfo>>') for 4th argument
    StorageDeltaMerge(
    ^
xxxx/dbms/src/Storages/StorageDeltaMerge.h:44:7: note: candidate constructor (the implicit copy constructor) not viable: requires 1 argument, but 8 were provided
class StorageDeltaMerge
      ^
[ 95%] Building CXX object dbms/CMakeFiles/gtests_dbms.dir/src/Storages/Page/V2/tests/gtest_data_compactor.cpp.o
In file xxxxxx/dbms/src/Flash/Management/tests/gtest_manual_compact.cpp:15:
In file xxxxxx/dbms/src/Common/FailPoint.h:16:
In file xxxxxx/dbms/src/Common/Exception.h:17:
In file xxxxxx/dbms/src/Common/StackTrace.h:17:
In file xxxxxx/dbms/src/Common/FmtUtils.h:17:
In file xxxxxx/libs/libcommon/include/common/StringRef.h:19:
In file xxxxxx/libs/libcommon/include/common/types.h:18:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.3.sdk/usr/include/c++/v1/string:506:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.3.sdk/usr/include/c++/v1/string_view:175:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.3.sdk/usr/include/c++/v1/__string:57:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.3.sdk/usr/include/c++/v1/algorithm:643:
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.3.sdk/usr/include/c++/v1/memory:4031:5: error: static_assert failed due to requirement 'is_constructible<Local, char const (&)[8], char const (&)[8], char const (&)[11], std::__1::reference_wrapper<TiDB::TableInfo>, DB::ColumnsDescription, std::__1::shared_ptr<DB::IAST> &, int, DB::Context &>::value' "Can't construct object in make_shared"
    static_assert(is_constructible<_Tp, _Args...>::value, "Can't construct object in make_shared");
    ^             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
xxxx/libs/libcommon/include/ext/shared_ptr_helper.h:49:21: note: in instantiation of function template specialization 'std::__1::make_shared<Local, char const (&)[8], char const (&)[8], char const (&)[11], std::__1::reference_wrapper<TiDB::TableInfo>, DB::ColumnsDescription, std::__1::shared_ptr<DB::IAST> &, int, DB::Context &>' requested here
        return std::make_shared<Local>(std::forward<TArgs>(args)...);
                    ^
In file xxxxxx/dbms/src/Flash/Management/tests/gtest_manual_compact.cpp:15:
In file xxxxxx/dbms/src/Common/FailPoint.h:16:
In file xxxxxx/dbms/src/Common/Exception.h:17:
In file xxxxxx/dbms/src/Common/StackTrace.h:17:
In file xxxxxx/dbms/src/Common/FmtUtils.h:17:
In file xxxxxx/libs/libcommon/include/common/StringRef.h:19:
In file xxxxxx/libs/libcommon/include/common/types.h:18:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.3.sdk/usr/include/c++/v1/string:506:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.3.sdk/usr/include/c++/v1/string_view:175:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.3.sdk/usr/include/c++/v1/__string:57:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.3.sdk/usr/include/c++/v1/algorithm:643:
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.3.sdk/usr/include/c++/v1/memory:1935:9: error: no matching constructor for initialization of 'Local'
      : __value_(_VSTD::forward<_Args>(_VSTD::get<_Indexes>(__args))...) {}
        ^        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.3.sdk/usr/include/c++/v1/memory:2019:9: note: in instantiation of function template specialization 'std::__1::__compressed_pair_elem<Local, 1, false>::__compressed_pair_elem<char const (&)[8], char const (&)[8], char const (&)[11], std::__1::reference_wrapper<TiDB::TableInfo> &&, DB::ColumnsDescription &&, std::__1::shared_ptr<DB::IAST> &, int &&, DB::Context &, 0, 1, 2, 3, 4, 5, 6, 7>' requested here
        _Base2(__pc, _VSTD::move(__second_args),
        ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.3.sdk/usr/include/c++/v1/memory:3297:16: note: in instantiation of function template specialization 'std::__1::__compressed_pair<std::__1::allocator<Local>, Local>::__compressed_pair<std::__1::allocator<Local> &, char const (&)[8], char const (&)[8], char const (&)[11], std::__1::reference_wrapper<TiDB::TableInfo> &&, DB::ColumnsDescription &&, std::__1::shared_ptr<DB::IAST> &, int &&, DB::Context &>' requested here
            :  __data_(piecewise_construct, _VSTD::forward_as_tuple(__a),
               ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.3.sdk/usr/include/c++/v1/memory:4038:26: note: in instantiation of function template specialization 'std::__1::__shared_ptr_emplace<Local, std::__1::allocator<Local>>::__shared_ptr_emplace<char const (&)[8], char const (&)[8], char const (&)[11], std::__1::reference_wrapper<TiDB::TableInfo>, DB::ColumnsDescription, std::__1::shared_ptr<DB::IAST> &, int, DB::Context &>' requested here
    ::new(__hold2.get()) _CntrlBlk(__a2, _VSTD::forward<_Args>(__args)...);
                         ^
xxxx/libs/libcommon/include/ext/shared_ptr_helper.h:49:21: note: in instantiation of function template specialization 'std::__1::make_shared<Local, char const (&)[8], char const (&)[8], char const (&)[11], std::__1::reference_wrapper<TiDB::TableInfo>, DB::ColumnsDescription, std::__1::shared_ptr<DB::IAST> &, int, DB::Context &>' requested here
        return std::make_shared<Local>(std::forward<TArgs>(args)...);
                    ^
xxxx/libs/libcommon/include/ext/shared_ptr_helper.h:43:16: note: candidate constructor (the implicit copy constructor) not viable: requires 1 argument, but 8 were provided
        struct Local : T
               ^
3 errors generated.
make[3]: *** [dbms/CMakeFiles/gtests_dbms.dir/src/Flash/Management/tests/gtest_manual_compact.cpp.o] Error 1
make[3]: *** Waiting for unfinished jobs....
make[2]: *** [dbms/CMakeFiles/gtests_dbms.dir/all] Error 2
make[1]: *** [dbms/CMakeFiles/gtests_dbms.dir/rule] Error 2
make: *** [gtests_dbms] Error 2
```
### 2. What did you expect to see? (Required)

### 3. What did you see instead (Required)

### 4. What is your TiFlash version? (Required)
`master`

--- a/dbms/src/Flash/Management/tests/gtest_manual_compact.cpp
+++ b/dbms/src/Flash/Management/tests/gtest_manual_compact.cpp
@@ -38,8 +38,6 @@ extern const char pause_before_server_merge_one_delta[];
 
 namespace tests
 {
-
-
 class BasicManualCompactTest
     : public DB::base::TiFlashStorageTestBasic
     , public testing::WithParamInterface<DM::tests::DMTestEnv::PkType>
@@ -83,7 +81,7 @@ public:
         storage = StorageDeltaMerge::create("TiFlash",
                                             "default" /* db_name */,
                                             "test_table" /* table_name */,
-                                            std::ref(table_info),
+                                            table_info,
                                             ColumnsDescription{columns},
                                             astptr,
                                             0,

